### DISCUSSION TITLE
Passing environment variables to publicRuntimeConfig doesn't work

### DISCUSSION THREAD

**Discussion by kyma** [vercel/next.js#17197] [Help]
I can hard-code a runtime variable in next.config.js in publicRuntimeConfig and it works fine.  But when I try to pass an environment variable from process.env it's always undefined.

## To Reproduce

`    publicRuntimeConfig: {
        cmsBase: process.env.CMS_BASE,
    },
`

`
"start": "cross-env CMS_BASE=https://my-cms.com next start"
`

`
function createApolloClient() {
    const {publicRuntimeConfig} = getConfig();
    const cmsBase = publicRuntimeConfig ? publicRuntimeConfig.cmsBase : 'http://localhost:1337';
`

This means that deploying onto multiple environments using Docker/Kubernetes would require a separate build for each env. Is there a way to get this working?

**Comment by janfi** [comment]
Hello,

I have the same problem. publicRuntimeConfig defines during build time not in runtime. So I need re-build before start nextjs....not optimal !

Thanks for your help.

**Comment by janfi** [comment]
I understand my mistake.

https://nextjs.org/docs/api-reference/next.config.js/runtime-configuration
"A page that relies on publicRuntimeConfig must use getInitialProps to opt-out of Automatic Static Optimization. Runtime configuration won't be available to any page (or component in a page) without getInitialProps"

without getInitialProps, no runtime env variables.

**Comment by kyma** [comment]
even with getInitialProps the value is undefined if it's passed from process.env though

**Reply by Kenam18** [reply]
Hi @kyma , were you able to achieve process.env in runtime configurations. Any suggestion?
